### PR TITLE
Fixes fire alarm closing shutters for adjacent area doors

### PR DESCRIPTION
**Pull request by Cthulhu80**
# About the pull request

From the code this looks intentional.... is it even a bug? #4880.

# Explain why it's good for the game

having all adjacent shutters close in a given area just looks a bit silly. 

shutters

# Changelog

:cl:
fix: fixes adjacent shutters closing after a fire alarm is pulled.
/:cl:

### CODE REVIEW

**Comment by Huffie56** [comment]
so yes commenting out looking out work for some situation but on other we need code to look out for adjacent shutter to close.
image
let's imagine an extreme example where red is "area A"(red)and is surrounded by other areas with their shutter at their access point(orange).
when i push the fire alarm in area A code need to look to adjacent area to close orange shutter.

**Comment by Cthulhu80** [comment]
Said this on discord but, maybe instead we can instead have it so that all connected fire shutters to any adjacent shutter close as well. That way it doesn't look funky and closes all the nearby room shutters.

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -54,11 +54,11 @@
...
-	for(var/direction in GLOB.cardinals)
-		A = get_area(get_step(src,direction))
-		if(istype(A) && !(A in areas_added))
-			A.all_doors.Add(src)
-			areas_added += A
+	// for(var/direction in GLOB.cardinals)  why are we even doing this?!
+	// 	A = get_area(get_step(src,direction))
+	// 	if(istype(A) && !(A in areas_added))
+	// 		A.all_doors.Add(src)
+	// 		areas_added += A
```
Comment (Drulikar) [inline code/game/machinery/doors/firedoor.dm:61]:
```suggestion
```
Just remove it.

File: code/game/machinery/doors/firedoor.dm [hunk 2]
```diff
@@ -54,11 +54,11 @@
 	A.all_doors.Add(src)
 	areas_added = list(A)
```
Comment (Drulikar) [inline code/game/machinery/doors/firedoor.dm:55]: Refactor out the need for the list `areas_added`

**Review by Drulikar — CHANGES_REQUESTED** [review]

**Review by fira — APPROVED** [review]
probably intended for atmos so who cares

**Review by Drulikar — CHANGES_REQUESTED** [review]
Works but it is likely desirable to just remove hallways fire alarms because there's just going to still be situations like this where the hallway is just a gigantic region so it having a firealarm affects things super far away - and even then it doesn't share fire doors that are shared by a different area: 
image
image

Technically in this situation it ought to have closed the doors in "Port Fore Lifeboat Fuel Pump" and "Medical Upper" (and maybe others because this area spans a good distance) to be a proper fire barrier. So if you opt to not remove hallway firealarms, then maybe consider adding the old code back but have it fire alarm that entire zone (but no further than it?) but I really do not expect this to work well. Technically the old code would have handled this situation better but obvious that has its own flaws. Alternatively we break up the areas more, and we add a way for firedoors to be linked to multiple areas in sets.

Also the fire alarms themselves don't change state visually in any way (most code bases change them to a flashing red state). We do have a state for this, just this code isn't setting it. 
image

**Comment by Cthulhu80** [comment]
> Works but **it is likely desirable to just remove hallways fire alarms** because there's just going to still be situations like this where the hallway is just a gigantic region so it having a firealarm affects things super far away - and even then it doesn't share fire doors that are shared by a different area: 
> 
> image
> 
> image
> 
> 
> 
> Technically in this situation it ought to have closed the doors in "Port Fore Lifeboat Fuel Pump" and "Medical Upper" (and maybe others because this area spans a good distance) to be a proper fire barrier. So if you opt to not remove hallway firealarms, then maybe consider adding the old code back but have it fire alarm that entire zone (but no further than it?) but I really do not expect this to work well. Technically the old code would have handled this situation better but obvious that has its own flaws. Alternatively we break up the areas more, and we add a way for firedoors to be linked to multiple areas in sets.
> 
> 
> 
> **Also the fire alarms themselves don't change state visually in any way (most code bases change them to a flashing red state). We do have a state for this, just this code isn't setting it.** Be sure it changes the state of all firealarms in an area if its possible for multiple firealarms to exist in a single area, and be sure it returns back to whatever code green/blue/red state. 
> 
> image
> 
> 

Sure yeah I can do that, @Huffie56, if u want to remove the hallway fire alarms and shutters. I can finish up the pr after. Feel free to commit your changes to this pr.

**Comment by Drulikar** [comment]
I'll note its probably fine to leave the shutters in hallways - I'm okay with there being no current way to deploy them. Its just this implementation off fire alarms in hallways isn't desirable to have them deploy so if we only have fire alarms in rooms it should work better with this implementation. But maybe maintainer will think we should find a better code solution to this.

**Comment by Cthulhu80** [comment]
> I'll note its probably fine to leave the shutters in hallways - I'm okay with there being no current way to deploy them. Its just this implementation of fire alarms in hallways isn't desirable to have them deploy so if we only have fire alarms in rooms it should work better with this implementation. But maybe maintainer will think we should find a better code solution to this.

I think the optimal solution would have to involve some sort of mapping, which again I don't know how to do. @Huffie56, what do you think?

**Comment by Huffie56** [comment]
My personal opinion is that removing current check is fine.

we also need to remind that their isn't only hallway that share emergency shutters.
that's why we had the code check for adjacent area in the first place...

one of the most used example is med bay and the OR's when you pull for treatment center room you want all the emergency for OR's to go down to...

just test in medbay and if you find this ok then go....

if i am not  very clear apology i am a bit tired...

**Comment by Cthulhu80** [comment]
I'm going to be gone for a bit @Huffie56, if u want to take this bug over. Closing the pr.

**Pull request closed without merging**